### ISSUE TITLE
PosterButton Context Menus

### ISSUE THREAD

**Issue by JPKribs** [jellyfin/Swiftfin#1157]
**Describe the feature you'd like**

I would be happy to work on this but I wanted to make a request here to make sure I wasn't stepping on anyone's toes before I started in earnest. The goal for this feature would be to copy item options to be accessible from the pagingLibraryView instead of needing to open them up to access the settings. This would be in addition to options on the ItemView. 

An example of this would be going over your list of movies and realizing that one was already watched so you can mark it as played without having to click into it. 

### iOS / iPadOS
1. Long press on a Movies, Series, or Episode
2. Get all options available for selection without entering the actual Media Item (Play, Shuffle, Played, Favorite, etc.)

### tvOS
1. Hold down the select button on a Movies, Series, or Episode
2. Get Non-Playback options (Favorite & Mark as Played)
-----
1. Hold down the play button on a Movies, Series, or Episode
2. Get Playback options (Play / Play from Beginning / Resume & Shuffle)

### Example Images

**Series**

<img width="232" alt="Screenshot 2024-07-23 at 14 00 55" src="https://github.com/user-attachments/assets/603e5d1e-617f-4a07-9e94-39323e2db8f1">

**Movie**

<img width="260" alt="Screenshot 2024-07-23 at 14 00 28" src="https://github.com/user-attachments/assets/9cd661a5-afe2-4b99-8a97-fed7d184c1c7">

**Additional context**
I was looking at this on my end and I already have some of these actions figured out for iOS. I see that there was previously some work for the resume vs play from beginning that I can help sort out first. Shuffle and Playlists are just filler for now as more of a down-the-road option along with downloads. I already have the Play (For Movies), Favorite, and Mark as Played options working but they don't update the posterButton so I would still need to figure out how to refresh that.

**Comment by LePips** [comment]
Yes, I've always wanted more context menu options. The main thing that has prevented it in the past from attaching it to every item is just how to update the item throughout the app. This should be possible now though by just having us multi-purpose the notification for an item being updated for view models that hold an `OrderedSet` of items.

**Comment by LePips** [comment]
Considering closed as #2075 has implemented the architecture for context menus. More options can be added.
